File: cpp/FizzBuzz/FizzBuzz/FizzBuzz.cpp
#include "FizzBuzz.h"


FizzBuzz::FizzBuzz(void)
{
}

FizzBuzz::~FizzBuzz(void)
{
}

std::string FizzBuzz::answer(int number)
{
	std::stringstream answer;
	
	answer << fizz(number) << buzz(number);
	if (answer.str() == "")
		answer << number;

	return answer.str();
}

std::string FizzBuzz::fizz(int number)
{
	return IsDivisibleByThree(number) ? "fizz" : "";
}

std::string FizzBuzz::buzz(int number)
{
	return IsDivisibleByFive(number) ? "buzz" : "";
}

bool FizzBuzz::IsDivisibleByThree(int number)
{
	return number != 0 && number % 3 == 0;
}

bool FizzBuzz::IsDivisibleByFive(int number)
{
	return number != 0 && number % 5 == 0;
}

File: FizzBuzz/csharp/FizzBuzzKata@SoC2010KickOff/FizzBuzzKata/FizzBuzz.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace FizzBuzzKata
{
    public class FizzBuzz
    {
        public String GetAnswer(int number)
        {
            if (IsFizzBuzz(number)) return "fizzbuzz";
            if (IsFizz(number)) return "fizz";
            if (IsBuzz(number)) return "buzz";
            
            return Convert.ToString(number); 
        }

        private Boolean IsFizz(int number)
        {
            return number % 3 == 0 && number != 0;
        }

        private Boolean IsBuzz(int number)
        {
            return number % 5 == 0 && number != 0;
        }

        private Boolean IsFizzBuzz(int number)
        {
            return number % 15 == 0 && number != 0;
        }
    }
}
